dev-guide/test_framework.md

```diff
@@ -0,0 +1,219 @@
+---
+layout: index
+title: GCAM Testing Framework
+gcam-version: v5.2
+devguide: dev-guide.html
+---
+## GCAM Testing Framework
+This page is here to serve as guidance for GCAM developers who may need to update or extend the GCAM Testing Framework.  Note the GCAM testing framework is meant to automate many of the steps developers must do to verify their changes do not break GCAM on any of the platforms GCAM is known to run on.  The services we use to do this are provided internal to PNNL, as noted below, therefore can not be used outside of the network.
```

Comment:
Are there plans to build a corresponding test suite for the publicly available `gcam-core` using Travis-CI or the like?

Reply:
That is a good question.  Thus far was have taken the approach that we will accept external user contributions as is (except maybe to ensure it works as intended and maybe basic fixes) and pull their changes over to Bitbucket where we will put in the additional leg work to get it to the point that it can be included in `master`.  (The idea being to maximize user's willingness to contribute)

From that perspective it is not clear what additional benefits of including test on the external repository as will will give us.  But there is nothing stopping us from doing so I suppose.  The only question I have is what sort of limits does Travis-CI have given GCAM is relatively slow and a memory hog.

Reply:
@pralitp We can table the open test suite conversation for now and see if there ever is a need in the future.  Thanks!